accounts-db/src/accounts_index.rs

```diff
@@ -485,31 +487,18 @@ pub struct AccountsIndexRootsStats {
     pub clean_dead_slot_us: u64,
 }
 
-pub struct AccountsIndexIterator<'a, T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> {
+type RangeItemVec<T> = Vec<(Pubkey, AccountMapEntry<T>)>;
+
+struct AccountsIndexIteratorInner<'a, T: IndexValue, U: DiskIndexValue + From<T> + Into<T>> {
     account_maps: &'a LockMapTypeSlice<T, U>,
     bin_calculator: &'a PubkeyBinCalculator24,
     start_bound: Bound<Pubkey>,
     end_bound: Bound<Pubkey>,
-    is_finished: bool,
```

Comment:
These two fields is only needed for "SorteIterator". so they are moved out of the shared "inner" class.